File: tests/data-conversion/expression/express.cpp
#include "runtime/string.h"

namespace {

void TestExpression() {
  if (true) {}
  std::optional<bool> optionalBoolean;
  if (compilets::IsTrue(optionalBoolean)) {}
  if (compilets::IsTrue(optionalBoolean) || 2 > 1) {}
  if (1 > 2) {}
  if (compilets::String(u"1") > u"2") {}
  if (compilets::StrictEqual(compilets::String(u"1"), u"1")) {}
}

}  // namespace

File: tests/data-conversion/expression/express.ts
function TestExpression() {
  if (true) {}
  let optionalBoolean: boolean | undefined;
  if (optionalBoolean) {}
  if (optionalBoolean || 2 > 1) {}

  if (1 > 2) {}
  if ('1' > '2') {}
  if ('1' === '1') {}
}
